crates/corro-agent/src/agent/handlers.rs

```diff
@@ -401,45 +376,127 @@ pub async fn handle_changes(
     mut rx_changes: CorroReceiver<(ChangeV1, ChangeSource)>,
     mut tripwire: Tripwire,
 ) {
+    const MIN_CHANGES_CHUNK: usize = 2000;
+    let mut queue: VecDeque<(ChangeV1, ChangeSource)> = VecDeque::new();
     let mut buf = vec![];
     let mut count = 0;
 
+    const MAX_CONCURRENT: usize = 3;
```

Comment:
It could make sense to update this number based on the capacity of the `rx_changes` channel

Reply:
I did wonder about that :) I wouldn't bother just yet but I think it's a good idea.